docs/source/en/quantization/quant_bnb.md

```diff
@@ -0,0 +1,486 @@
+    mem_usage = memory_dict[key]
+    ax.text(
+        0.5, 0.01,
+        f"{key}\nMemory: {mem_usage:.2f} GB",
+        transform=ax.transAxes, color='white', fontsize=12,
+        ha='center', va='bottom'
+    )
+
+# Plot the memory usage trend line
+ax_line = fig.add_subplot(gs[1, :])
+ax_line.plot(keys, memory_values, marker='o', linestyle='-', color='orange')
+ax_line.set_xlabel('Quantization Type')
+ax_line.set_ylabel('Memory Usage (GB)')
+ax_line.set_title('Reduction in Memory Usage with Different Quantization Methods')
+ax_line.grid(True)
+
+# Annotate each point with memory usage
+for i, mem in enumerate(memory_values):
+    ax_line.annotate(
+        f"{mem:.2f} GB", (keys[i], mem), textcoords="offset points",
+        xytext=(0,10), ha='center'
+    )
+
+# Adjust x-axis labels to match the images
+ax_line.set_xticks(keys)
+ax_line.set_xticklabels(keys)
+
+# Adjust layout
+plt.tight_layout()
+plt.show()
+```
+
+![nf4 image](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/diffusers/quant-bnb/comparison.png)
+
+This visualization helps us see how different quantization methods reduce memory usage and how they
+affect the generated images. Generally, lower-bit quantization reduces memory consumption but may
+introduce artifacts or degrade image quality. However, as you can see, the images generated with
+4-bit and NF4 quantization are still quite acceptable, especially considering the significant memory
+savings.
+
+## Training LoRAs with Quantized Weights
+
+If you think quantization only helps with **inference**, you're in for a surprise. The `diffusers`
+library supports training of LoRAs (Low-Rank Adaptations) with quantized weights as well.
+
+Due to space constraints, we won't dive into the detailed process here. However, you can find an
+experimental setup and more information in the resource linked below.
+
+- An experimental setup [3] for training LoRAs with quantized weights for FLUX.1-dev.
+
+## Conclusion
+
+Quantization is a powerful technique for reducing the memory footprint of diffusion models without
+significantly compromising their performance. By applying 8-bit, 4-bit, or NF4 quantization, you can
+run large models like FLUX.1-dev on hardware with limited VRAM, such as a free-tier Google Colab instance.
+This opens up new possibilities for deploying advanced models on consumer-grade devices.
+
+Whether you're interested in inference or training, quantization provides a practical solution for
+resource constraints. We encourage you to experiment with these techniques and see how they can
+benefit your projects.
+
+## References
+
+- [1] [bitsandbytes GitHub Repository](https://github.com/bitsandbytes-foundation/bitsandbytes)
+- [2] [FLUX.1-dev Model on Hugging Face](https://huggingface.co/black-forest-labs/FLUX.1-dev)
+- [3] [Training LoRAs with Quantized Weights for FLUX.1-dev](https://github.com/huggingface/diffusers/tree/main/examples/research_projects/flux_lora_quantization)
```

Comment:
You can just link directly where you need to instead of adding these footnote references :)

```suggestion
```